crates/oxc_minifier/src/ast_passes/peephole_replace_known_methods.rs

```diff
@@ -155,6 +157,38 @@ impl PeepholeReplaceKnownMethods {
 
         return Some(ctx.ast.expression_from_string_literal(ctx.ast.string_literal(span, result)));
     }
+
+    fn try_fold_string_char_code_at<'a>(
+        span: Span,
+        call_expr: &CallExpression<'a>,
+        string_lit: &StringLiteral<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) -> Option<Expression<'a>> {
+        let char_at_index: Option<f64> = match call_expr.arguments.first() {
+            Some(Argument::NumericLiteral(numeric_lit)) => Some(numeric_lit.value),
+            Some(Argument::UnaryExpression(unary_expr))
+                if unary_expr.operator == UnaryOperator::UnaryNegation =>
+            {
+                let Expression::NumericLiteral(numeric_lit) = &unary_expr.argument else {
+                    return None;
+                };
+                Some(-(numeric_lit.value))
+            }
+            None => None,
+            _ => return None,
+        };
```

Comment:
```rust
        let char_at_index = call_expr.arguments.first().and_then(|idx| ctx.get_number_value(idx))
```
Similar approches, you don't need to handle it manually.

Reply:
fixed